types/enzyme/index.d.ts

```diff
@@ -447,6 +447,11 @@ export class ShallowWrapper<P = {}, S = {}, C = Component> {
      * Returns a wrapper with the direct parent of the node in the current wrapper.
      */
     parent(): ShallowWrapper<any, any>;
+
+    /**
+     * Returns a wrapper of the node rendered by the provided render prop.
+     */
+    renderProp(prop: string): (params: any) => ShallowWrapper<any, any>;
```

Comment:
We should be able to strongly type the params here to match the render prop, since we expect it to be a callable function. Something like:
```ts
renderProp<PropName extends keyof P>(prop: PropName): (params: Parameters<P[PropName]>) => ShallowWrapper<any, any>;
```
where Parameters is defined as `type Parameters<T> = T extends (...args: infer A) => any ? A : never`. This was shipped as a built-in helper in TS 3.1 but I don't know if we should bump the minimum version to that yet.